android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/MetadataResolver.java

```diff
@@ -236,6 +249,17 @@ public void onErrorResponse(VolleyError volleyError) {
     );
   }
 
+  public static boolean checkIfMetadataContainsSortingScores(Collection<UrlMetadata> urlMetadataCollection) {
```

Comment:
Checking items on the list to make sure a flag is not On doesn't sound like
a good programming practice. Of course you can make assumptions to
implement hacky methods, but that doesn't mean you should. The server
should explicitly communicate whether or not a flag is on.

On Tue Jan 27 2015 at 5:55:58 PM Hoà V. DINH <notifications@github.com>
wrote:

> In
> android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/MetadataResolver.java
> <https://github.com/google/physical-web/pull/277#discussion_r23660652>:
>
> > @@ -236,6 +249,17 @@ public void onErrorResponse(VolleyError volleyError) {
> >      );
> >    }
> >
> > +  public static boolean checkIfMetadataContainsSortingScores(Collection<UrlMetadata> urlMetadataCollection) {
>
> 1. Checking the whole list is certainly not big trouble since the sorting
> itself will be slower anyway. 2. We can make assumptions. If we get a score
> for at least one item. We can assume the feature is enabled on the server
> and assume that every metadata has a valid score.
> … <#msg-f:1491505154513765569_>
> -- Hoa V. Dinh
> On Tuesday, January 27, 2015 at 5:43 PM, Giovanni Ortuno wrote: In
> android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/MetadataResolver.java
> (https://github.com/google/physical-web/pull/277#discussion_r23660131): >
> @@ -236,6 +249,17 @@ public void onErrorResponse(VolleyError volleyError) {
> > ); > } > > + public static boolean
> checkIfMetadataContainsSortingScores(Collection<UrlMetadata>
> urlMetadataCollection) { Since there are two ways of sorting you have to
> know if the whole list hasScore or not. To do that you still need to check
> all elements. — Reply to this email directly or view it on GitHub (
> https://github.com/google/physical-web/pull/277/files#r23660131).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/physical-web/pull/277/files#r23660652>.
>